### PR TITLE
Support tuples and atoms as generators

### PR DESCRIPTION
**Pull request by whatyouhide**
Closes https://github.com/whatyouhide/stream_data/issues/7.

### CODE REVIEW

File: lib/stream_data.ex [hunk 1]
```diff
@@ -82,6 +82,26 @@ defmodule StreamData do
...
+  Some Elixir types are implicitly converted to `StreamData` generators when
+  used in property testing or composed. These types are:
+
+    * atoms - they generated themselves. For example, `:foo` is equivalent to
+      `StreamData.constant(:foo)`.
+
+    * tuples of generators - they generated tuples where each value is a value
+      generated by the corresponding generator, exactly like described in
+      `tuple/1`. For example, `{StreamData.int(), StreamData.boolean()}`
+      generates things like `{10, false}`.
```
Comment (josevalim) [inline lib/stream_data.ex]: s/things/entries?

File: test/stream_data_test.exs [hunk 2]
```diff
@@ -15,6 +15,20 @@ defmodule StreamDataTest do
     end)
   end
 
+  test "atoms are generators" do
+    for_many(map(:foo, &(&1)), fn term ->
+      assert term == :foo
+    end)
+  end
+
+  test "tuples are generators" do
```
Comment (josevalim) [inline test/stream_data_test.exs]: Should we add a test of a tuple with atoms inside?

**Comment by josevalim** [comment]
Should we add a function that converts a data structure to a generator if possible? I know today it can be done as simple as `Stream.Data.map(arg, & &1)` though.

**Comment by whatyouhide** [comment]
@josevalim fixed the two comments, thanks for pointing out. 

> Should we add a function that converts a data structure to a generator if possible? I know today it can be done as simple as Stream.Data.map(arg, & &1) though.

Yes it can and I would wait with a function that does that. Maybe we can document that you can use `map/2` to turn into a generator. Thoughts?

**Comment by josevalim** [comment]
No need to document it explicitly. Let's wait until the question shows up, then we can discuss it accordingly.

**Pull request merged**